src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskCache.cs

```diff
@@ -13,6 +13,8 @@ internal static class TaskCache
         internal static readonly Task<bool> s_trueTask = CreateCacheableTask(result: true);
         /// <summary>A cached Task{Boolean}.Result == false.</summary>
         internal static readonly Task<bool> s_falseTask = CreateCacheableTask(result: false);
+        /// <summary>A cached Task{String}.Result == "".</summary>
+        internal static readonly Task<string> s_emptyStringTask = CreateCacheableTask(result: "");
```

Comment:
We optimize for common scenarios. I don't think that reading from `xReader.Null` is common and it should not be on a hot path in a well designed application.

Caching tasks like that is not for free (it prolongs the time it takes to initialize the static fields), please remove the cached task.

Reply:
I'm assuming you'd prefer to go back to `Task.FromResult("")` in the method body, not even using a `static readonly` field in `NullStreamReader` itself; I will revert this change (which was suggested at https://github.com/dotnet/runtime/pull/64301/files/426c0414a3a7ad2759a44acb9fd5abbb9c6505e6#r792278679).